From e4bca69d735e13ac75e9749eb739b6c070c7ad6d Mon Sep 17 00:00:00 2001
From: Jean-Daniel Cryans <jdcryans@apache.org>
Date: Sat, 8 Dec 2012 00:46:56 +0000
Subject: [PATCH 056/202] HBASE-7300  HbckTestingUtil needs to keep a static executor to lower the number of threads used

Reason: Improvement
Author: Jean-Daniel Cryans
Ref: CDH-9404

git-svn-id: https://svn.apache.org/repos/asf/hbase/trunk@1418571 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit ae692efab78a929494b5e2decaa0129338e7124d)
---
 .../hadoop/hbase/util/hbck/HbckTestingUtil.java    |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/test/java/org/apache/hadoop/hbase/util/hbck/HbckTestingUtil.java b/src/test/java/org/apache/hadoop/hbase/util/hbck/HbckTestingUtil.java
index 299ae6e..3434021 100644
--- a/src/test/java/org/apache/hadoop/hbase/util/hbck/HbckTestingUtil.java
+++ b/src/test/java/org/apache/hadoop/hbase/util/hbck/HbckTestingUtil.java
@@ -30,6 +30,7 @@ import org.apache.hadoop.hbase.util.HBaseFsck;
 import org.apache.hadoop.hbase.util.HBaseFsck.ErrorReporter.ERROR_CODE;
 
 public class HbckTestingUtil {
+  private static ExecutorService exec = new ScheduledThreadPoolExecutor(10);
   public static HBaseFsck doFsck(
       Configuration conf, boolean fix) throws Exception {
     return doFsck(conf, fix, null);
@@ -44,7 +45,7 @@ public class HbckTestingUtil {
       boolean fixMeta, boolean fixHdfsHoles, boolean fixHdfsOverlaps,
       boolean fixHdfsOrphans, boolean fixTableOrphans, boolean fixVersionFile,
       String table) throws Exception {
-    HBaseFsck fsck = new HBaseFsck(conf);
+    HBaseFsck fsck = new HBaseFsck(conf, exec);
     fsck.connect();
     fsck.setDisplayFullReport(); // i.e. -details
     fsck.setTimeLag(0);
@@ -71,7 +72,6 @@ public class HbckTestingUtil {
    */
   public static HBaseFsck doHFileQuarantine(Configuration conf, String table) throws Exception {
     String[] args = {"-sidelineCorruptHFiles", "-ignorePreCheckPermission", table};
-    ExecutorService exec = new ScheduledThreadPoolExecutor(10);
     HBaseFsck hbck = new HBaseFsck(conf, exec);
     hbck.exec(exec, args);
     return hbck;
-- 
1.7.0.4

